src/main/java/server/request/RequestParser.java

```diff
@@ -1,29 +1,49 @@
 package main.java.server.request;
 
+import java.util.HashMap;
+
 public class RequestParser {
     private static final int REQUEST_METHOD_INDEX = 0;
     private static final int REQUEST_PATH_INDEX = 1;
     private static final int REQUEST_VERSION_INDEX = 2;
 
     public Request create(String incomingRequest) throws RequestParseException {
         try {
-            String[] segmentedStatusLine = incomingRequest.split("\r\n")[0].split(" ");
-            StatusLine requestStatusLine = buildStatusLine(segmentedStatusLine);
-            return new Request(requestStatusLine);
+            StatusLine requestStatusLine = buildStatusLine(incomingRequest);
+            Headers requestHeaders = buildHeaders(incomingRequest);
+            return new Request(requestStatusLine, requestHeaders);
         } catch (Exception e) {
             throw new RequestParseException(e);
         }
     }
 
-    private StatusLine buildStatusLine(String[] segmentedStatusLine) {
+    private StatusLine buildStatusLine(String incomingRequest) {
+        String[] segmentedStatusLine = incomingRequest.split("\r\n")[0].split(" ");
+
         StatusLine requestStatusLine = new StatusLine();
+
         requestStatusLine.setMethod(getMethod(segmentedStatusLine));
         requestStatusLine.setPath(getPath(segmentedStatusLine));
         requestStatusLine.setVersion(getVersion(segmentedStatusLine));
 
         return requestStatusLine;
     }
 
+    private Headers buildHeaders(String incomingRequest) {
+        String[] segmentedHeaders = incomingRequest.split("\r\n\r\n")[0].split("\n");
```

Comment:
Feels like there's a lot of detailed knowledge about proper request formatting build into the `split` calls in this namespace. Any way to extract things behind intention revealing variables or functions?